### PR TITLE
pass the captured environment variables to rtpSpawn

### PR DESCRIPTION
**Pull request by BaoshanPang**
capture_env() would capture all environment variables set before invoking a RTP process, I didn't realize it is different to what can get from sys::os::environ() when I removed it.

The reason I removed capture_env() beofore is I found the "envp" returned by it would be released when rtpSpawn() still need to use it.

Now I find out using as_ref() would make it keep alive.

This is to fix the issue found with test case test-panic-abort.rs which is reported in V7LIBC-1054.

r? @n-salim 
cc @UmeshKalappa

### CODE REVIEW

File: src/libstd/sys/vxworks/process/process_vxworks.rs [hunk 1]
```diff
@@ -52,10 +53,13 @@ impl Command {
                 t!(cvt(libc::chdir(cwd.as_ptr())));
             }
 
+            let envp = envp.as_ref().map(|c| c.as_ptr())
+                .unwrap_or_else(|| *sys::os::environ() as *const _);
```
[the remark this answers is not in the document]
Comment (BaoshanPang) [reply src/libstd/sys/vxworks/process/process_vxworks.rs]: In vxWorks os.rs setenv(), a lock is fecthed before calling libc::setenv, so I think we already did it.

[the remark this answers is not in the document]
Comment (BaoshanPang) [reply src/libstd/sys/vxworks/process/process_vxworks.rs]: added and used env_lock for vxWorks.

File: src/libstd/sys/vxworks/process/process_vxworks.rs [hunk 2]
```diff
@@ -52,10 +53,16 @@ impl Command {
                 t!(cvt(libc::chdir(cwd.as_ptr())));
             }
 
+            // ensure that access to the environment is synchronized
+            let _lock = sys::os::env_lock();
```
Comment (n-salim) [inline src/libstd/sys/vxworks/process/process_vxworks.rs]: Please explain to me how this solves the problem.

Comment (n-salim) [reply src/libstd/sys/vxworks/process/process_vxworks.rs]: Thanks for fixing the problem (two different notions of ENV_LOCK!).

**Pull request merged**